components/hab/src/command/studio/enter.rs

```diff
@@ -74,6 +75,19 @@ pub fn start(ui: &mut UI, args: &[OsString]) -> Result<()> {
         stdfs::create_dir_all(&artifact_path)?;
     }
 
+    let ssl_path = match henv::var(SSL_PATH_ENVVAR) {
+        Ok(p) => PathBuf::from(p),
+        Err(_) => {
+            let path = fs::cache_ssl_path(None::<&str>);
+            debug!("Setting {}={}", SSL_PATH_ENVVAR, path.display());
+            env::set_var(SSL_PATH_ENVVAR, &path);
```

Comment:
Setting the environment variable like this is a bit surprising, and potentially fragile. If we want to make sure there's a value for this environment variable set before we spawn the actual studio process, I think it would be better to set it in `inner::start`, where we manipulate the rest of the environment we're creating for the exec'd process.

I can see someone easily refactoring the environment code in there for better isolation, etc., and totally removing this SSL path in the process, since it was handled remotely.

Reply:
Good point, there's probably room to refactor all this, however I'd prefer to not do refactoring or make flow changes in this PR

Reply:
This is all new code, though.

Reply:
It's essentially exactly the same pattern as the code above is my point - I'm happy to open an issue to refactor as a separate PR

Reply:
I took a look at the code above this... yup, same pattern :scream: 

I think we really do need to refactor all this (separate PR is fine), and put some effort into wrapping up https://github.com/habitat-sh/habitat/issues/6314 as well.